### ISSUE TITLE
tree-sitter-cli cannot run emcc on windows

### ISSUE THREAD

**Issue by facekapow** [tree-sitter/tree-sitter#434]
I'm trying to set up WebAssembly compilation on Windows using tree-sitter-cli's `build-wasm` subcommand and even though I have emcc in my path (after running emsdk_env.bat), the CLI complains that "You must have either emcc or docker on your PATH to run this command".

After inspecting the code a bit, it appears that this is the issue:
https://github.com/tree-sitter/tree-sitter/blob/105b0ba52ce64e85eb2d6f09f5fe3ad508adeafb/cli/src/wasm.rs#L25

The problem is that when `emcc` is run by itself, it prints "emcc:WARNING: no input files" and exits with exit code 1 (which counts as failure).

I'd be happy to open up a pull request to fix this (by checking the output from emcc) if that's ok.

**My specs**
```
Windows 10
emcc (Emscripten gcc/clang-like replacement) 1.38.42 (commit da4900b83f6bce7511189b9307569d6c9c344e36)
tree-sitter 0.15.7 (5d43b58751ca4135a42e37b3b52f9bbea3055ab1)
```

**Comment by facekapow** [comment]
Actually, upon further inspection, it appears that Rust is actually unable to run the command (I'm unfamiliar with Rust and so I thought the `is_ok` method checks for exit code. This makes my issue even weirder, since emcc is most definitely in my path and should be executable by tree-sitter.

**Comment by Razzeee** [comment]
I can only recommend to use docker, I invested some hours into emcc without much success, then switched to using docker and it almost instantly worked.

**Comment by maxbrunsfeld** [comment]
Maybe on windows it needs to run it as ‘emcc.exe’?

**Comment by jbouecke** [comment]
I alos didn't get it to work on windows, and sinc I only own 'home' (not 'pro), docker is alos not quite ease. However, using the WSL (with debian) I was able to get the emcc to compile the WASM straght forward.

**Comment by sogaiu** [comment]
FWIW, it appears that there's a `emcc.bat`:
```
C:\Users\user\Desktop\tree-sitter-clojure>emcc.bat
emcc:WARNING: no input files
```

**Comment by sogaiu** [comment]
Tried putting the full path to the `.bat` file in the appropriate spots in:

https://github.com/tree-sitter/tree-sitter/blob/master/cli/src/wasm.rs#L25
https://github.com/tree-sitter/tree-sitter/blob/master/cli/src/wasm.rs#L26

Using the resulting `tree-sitter.exe`, had some success in creating a `.wasm` file.  Verified it worked via the `web-ui` subcommand.

FWIW, `Command` on Windows can be troublesome: https://github.com/rust-lang/rust/issues/37519

Not sure yet what a good resolution is.  Docker isn't desirable for everybody / all situations (and neither is WSL), so it would be nice to have this work without either.

(On a side note, the `.appveyor.yml` file was helpful in figuring out what is necessary to build on Windows.  Specifically, had the toolchain part wrong.)

**Comment by sogaiu** [comment]
[One of the comments in the aforementioned issue](https://github.com/rust-lang/rust/issues/37519#issuecomment-362276543) pointed out the [which](https://crates.io/crates/which) crate.

I gave this a try using `emcc.bat` as an argument to `which` and it seemed to work.

Pardon my newbie rust, but the gist of what I tried is (apart from arranging for `which` to be available):
```
if !force_docker && Command::new(which("emcc.bat").unwrap().to_str().unwrap()).output().is_ok() {
  command = Command::new(which("emcc.bat").unwrap().to_str().unwrap());
```

It's not checking for the platform, but just to give a sense of what seems to work.
